armory/datasets/README.md

```diff
@@ -0,0 +1,115 @@
+# Setup
+
+This is designed to be run in an armory pytorch container.
+Recommended use:
+```
+armory launch pytorch
+```
+Then work in that environment.
+
+# Structure
+
+The code exists in `armory/datasets`, with two subdirectories:
+- `standard` - non-adversarial datasets
+- `adversarial` - adversarial datasets
+
+# Create a new dataset for integration into armory
+
+See the [TFDS v4 CLI](https://www.tensorflow.org/datasets/cli).
+
+On the command line, to create a new standard dataset with name `"my_dataset"`, do:
+```
+cd armory/datasets/standard
+tfds new my_dataset
+```
+This will populate the `armory/datasets/standard/my_dataset` directory.
+
+At a minimum, you will need to fill in the `my_dataset.py` and `checksums.tsv` files.
+
+## Building and loading
+
+### TFDS standard dataset
+
+MNIST is a standard built-in TFDS dataset. To build this, do:
+```
+python -m armory.datasets.build mnist
+```
+
+To load it, you can do:
+```
+python -m armory.datasets.load mnist
```

Comment:
what's the intended use case in running `armory.datasets.load` as `__main__()`, simply to test whether it can load? I ask because running that script will load the dataset and do nothing. The interpreter example below seems more likely to resemble the use case I'd think, but wanted to check if I was missing something

Reply:
I was thinking mainly for testing builds in a scripted fashion. I think that if you load a number of datasets sequentially in the same python instantiation, you may hit memory issues.